### DISCUSSION TITLE
Get secret from hosting action repo

### DISCUSSION THREAD

**Discussion by MauricioRobayo** [community/community#26141] [Actions]
<p>Is it possible to get a secret from the repository an action is hosted?</p>
<p>Not getting a secret from the repository where the action is run, but from the repo where the action being used is hosted, so that workflows using the action can use the secret from the action repo.</p>
<p>I want to run some tests for anyone using my action, but I don’t want the users of the action to see the tests, just the output. So I was thinking on hosting an action and using a secret on the action repo to get the tests from a private repo, to run them on a workflow by other users without letting them see the tests cases.</p>
<p>Is something like this possible?</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Hi <a class="mention" href="/mauriciorobayo">@mauriciorobayo</a>,</p>
<p>That’s not possible. Secrets are only available from the repository that owns the workflow.</p>
<p>I can’t think of a way you could accomplish this securely. Anything embedded in the Action would be available for anyone to see.</p>
<p>-Mike </p>
